src/molecules/Select/Select.test.tsx

```diff
@@ -1,111 +1,161 @@
 
-  it.todo('closes the menu when keydown escape');
+    const options = queryAllByTestId('select-custom-opt');
+    expect(options).toHaveLength(4);
 
-  it('returns the options value to the click select click handler onClick', () => {
-    let wrapper = setup();
-    let button = wrapper.find('button');
-    button.simulate('click');
-    let options = wrapper.find(Option).first();
-    options.simulate('click');
-    expect(mockClick).toHaveBeenCalledWith('london');
+    expect(options[0]).toHaveTextContent('UK');
+    expect(options[1]).toHaveTextContent('UK');
+    expect(options[2]).toHaveTextContent('UK');
+    expect(options[3]).toHaveTextContent('US');
   });
 
-  it('calls the onClick options handler when an option is clicked', () => {
-    const setup = () =>
-      mountWithTheme(
-        <Select placeholder={'I am placeholder'}>
-          {() =>
-            citiesData.map((cityData, index) => (
-              <Option
-                key={index}
-                value={cityData.value}
-                onClick={mockClickOption}
-              />
-            ))
-          }
-        </Select>
-      );
-
-    let wrapper = setup();
-    let button = wrapper.find('button');
-    button.simulate('click');
-    let options = wrapper.find(Option);
-    options.first().simulate('click');
-    expect(mockClickOption).toHaveBeenCalled();
+  it('executes a callback and updates the label when an option is selected', () => {
+    const mockClick = jest.fn();
+    const { queryByTestId, queryAllByTestId } = setup({
+      onChange: mockClick,
+      getOptionLabel: (opt: any) => `${opt.title}, ${opt.country}`
+    });
+
+    fireEvent.click(queryByTestId('select')!);
+    fireEvent.click(queryAllByTestId('select-option')[0]);
+
+    expect(mockClick).toHaveBeenCalledTimes(1);
+    expect(mockClick).toHaveBeenCalledWith(cities[0]);
+
+    expect(queryByTestId('select-input')).toHaveValue('London, UK');
   });
 
-  it('collapses the open menu when an option is clicked', () => {
-    let wrapper = setup();
-    let button = wrapper.find('button');
-    button.simulate('click');
-    let options = wrapper.find(Option);
-    expect(options.length).toBe(3);
-    options.first().simulate('click');
-    let optionUpdated = wrapper.find(Option);
-    expect(optionUpdated.length).toBe(0);
+  it('executes a callback and updates the label when an option is selected and a defaultValue is given', () => {
+    const mockClick = jest.fn();
+    const { queryByTestId, queryAllByTestId } = setup({
+      onChange: mockClick,
+      defaultValue: cities[2]
+    });
+
+    expect(queryByTestId('select-input')).toHaveValue('Bristol');
+
+    fireEvent.click(queryByTestId('select')!);
+    fireEvent.click(queryAllByTestId('select-option')[1]);
+
+    expect(mockClick).toHaveBeenCalledTimes(1);
+    expect(mockClick).toHaveBeenCalledWith(cities[1]);
+
+    expect(queryByTestId('select-input')).toHaveValue('Cardiff');
   });
 });
```

Comment:
Looking at the difference between old testing vs RTL - this is soooo much better